src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -1,168 +1,64 @@
-                }
-
-                // add a data point to the OMF message body
-                OSIsoftServer.addPointToOMFDataMessage(tagValue, timeStamp);
-                break;
-              case OSIsoftConfig.piwebapi:
-                OSIsoftServer.addPointToBatch(
-                    ((Tag) OSIsoftConfig.tags.get(tagIndex)),
-                    (DataPoint) ((ArrayList) dataPoints.get(tagIndex)).get(dataPointIndex));
 
-                break;
-              default:
-                Logger.LOG_ERR(comsErrMsg);
-                break;
-            }
+    while (true) {
 
-            pointsAdded++;
-          } else {
-            Logger.LOG_ERR("Null data point encountered");
-          }
-        }
-        switch (communicationType) {
-          case OSIsoftConfig.omf:
-            OSIsoftServer.addContainerEndToOMFDataMessage();
-            break;
-          case OSIsoftConfig.piwebapi:
-            OSIsoftServer.startBatch();
-            break;
-          default:
-            Logger.LOG_ERR(comsErrMsg);
-            break;
-        }
-      }
+      String payload = payloadHolder.getNextPayload();
 
-      switch (communicationType) {
-        case OSIsoftConfig.omf:
-          OSIsoftServer.endOMFDataMessage();
-          break;
-        case OSIsoftConfig.piwebapi:
-          OSIsoftServer.endBatch();
-          break;
-        default:
-          Logger.LOG_ERR(comsErrMsg);
-          break;
-      }
+      // If there are any payloads to send
+      if (payload.length() > 0) {
+        Logger.LOG_DEBUG("[DataPoster] Sending completed payload");
 
-      if (pointsAdded > 0) {
-        Logger.LOG_DEBUG("Sending " + pointsAdded + " points to server");
-        // Post the tags to the server
         boolean retval = false;
-
-        switch (communicationType) {
+        switch (OSIsoftConfig.getCommunicationType()) {
           case OSIsoftConfig.omf:
-            retval = OSIsoftServer.postOMFBatch();
+            retval = OSIsoftServer.postOMFBatch(payload);
             break;
           case OSIsoftConfig.piwebapi:
-            retval = OSIsoftServer.postBatch();
+            retval = OSIsoftServer.postBatch(payload);
             break;
           default:
-            Logger.LOG_ERR(comsErrMsg);
+            Logger.LOG_SERIOUS(OSIsoftConfig.COM_ERR_MSG);
             break;
         }
 
-        if (retval) {
-          // Remove all the points that were posted
-          for (int tagIndex = 0; tagIndex < OSIsoftConfig.tags.size(); tagIndex++) {
-            ((Tag) OSIsoftConfig.tags.get(tagIndex))
-                .removeDataPoints((ArrayList) dataPoints.get(tagIndex));
-          }
+        if (!retval) {
+          Logger.LOG_WARN("[DataPoster] Unable to send payload to OSIsoft");
         }
       }
 
-      // Tell the JVM that it should garbage collect soon
-      System.gc();
+      // thread finished
+      Thread.yield();
+      try {
+        int oneSecond = 1000;
+        Thread.sleep(oneSecond);

```

Comment:
```suggestion
        final int sleepTimeMs = 1000;
        Thread.sleep(sleepTimeMs);
```
putting the value in the variable name is not helpful and leads to more code maintenance if you need to change it.